crates/ibc/src/core/ics03_connection/handler/verify.rs

```diff
@@ -28,19 +28,17 @@ pub fn verify_proofs(
     )?;
 
     // If the message includes a client state, then verify the proof for that state.
-    if let Some(expected_client_state) = client_state {
-        verify_client_proof(
-            ctx,
-            height,
-            connection_end,
-            expected_client_state,
-            proofs.height(),
-            proofs
-                .client_proof()
-                .as_ref()
-                .ok_or_else(Error::null_client_proof)?,
-        )?;
-    }
+    verify_client_proof(
+        ctx,
+        height,
+        connection_end,
+        expected_client_state,
+        proofs.height(),
+        proofs
+            .client_proof()
+            .as_ref()
+            .ok_or_else(Error::null_client_proof)?,
+    )?;
 
     // If a consensus proof is attached to the message, then verify it.
     if let Some(proof) = proofs.consensus_proof() {
```

Comment:
Nit: I think it will be nice to assert here that the `consensus_proof` does exist and is not optional, but we can also do it in a separate PR. Related to #16.

Reply:
Good point; let's address it in #15.